processor.go

```diff
@@ -143,7 +143,11 @@ func (p *processor) handle(ctx *fh.RequestCtx) {
 		}
 	}
 
-	// Return 500 for any error
```

Comment:
I'd swap this block with the next one with error checking. This way it only gets executed if there are no errors.

Also i'd set the code to 204 and `results[0].body` to nil so that the body with an error would not get transferred to Prometheus.